muxers/yamux/src/lib.rs

```diff
@@ -41,23 +37,25 @@ use thiserror::Error;
 use yamux::ConnectionError;
 
 /// A Yamux connection.
-pub struct Muxer<S> {
+pub struct Muxer<C> {
     /// The [`futures::stream::Stream`] of incoming substreams.
-    incoming: S,
+    incoming: BoxStream<'static, Result<yamux::Stream, yamux::ConnectionError>>,
```

Comment:
Why `Box` the stream of streams? Not boxing would also resolve the need for `PhantomData`, right?

Reply:
I just inlined what we had before.

This and the phantom data will go away once update to latest yamux. I only added it to make sure it is a compatible API change later :)

`into_stream` returns an `impl Trait`. I guess we could inline the unfold operation and store an `Unfold` type?

Do you feel strongly?

Reply:
I am not sure the `Unfold` type is nameable at all because it contains an async block.

Reply:
> This and the phantom data will go away once update to latest yamux.

Great. Don't feel strongly. Let's keep as is.